addons/bestja_detailed_reports/models.py

```diff
@@ -0,0 +1,215 @@
+        domain=lambda self: [
+            ('parent.use_detailed_reports', '=', True),
+            ('detailed_reports', '=', False),
+            '|',
+            ('manager', '=', self.env.user.id),
+            ('organization.coordinator', '=', self.env.user.id)],
+        ondelete='restrict',
+    )
+    organization = fields.Many2one(
+        'organization',
+        string="Organizacja",
+        related='project.organization',
+    )
+    name = fields.Char(string="Nazwa projektu", related="project.name")
+    dates = fields.Char(string="Termin", compute="compute_project_dates", store=True)
+    state = fields.Selection(STATES, default='draft', string="Status:")
+    report_entries = fields.One2many('bestja.report_entry', inverse_name='detailed_report', string="Produkt")
+    tonnage = fields.Float(string="Tonaż (kg)", compute="compute_report_tonnage", store=True)
+    parent_project = fields.Many2one(
+        'bestja.project',
+        string="Projekt nadrzędny",
+        related='project.parent',
+        store=True,
+    )
+    interested_organization = fields.Many2one(
+        'organization',
+        string="Zainteresowane organizacje",
+        compute="compute_interested_organization",
+        store=True,
+    )
+    user_can_moderate = fields.Boolean(compute="compute_user_can_moderate")
+
+    @api.one
+    @api.depends('parent_project', 'project')
+    def compute_interested_organization(self):
+        level = self.project.organization.level
+        if level <= 1:
+            self.interested_organization = self.project.organization.id
+        else:
+            self.interested_organization = self.parent_project.organization.id
+
+    @api.one
+    @api.depends('report_entries')
+    def compute_report_tonnage(self):
+        """
+        For showing the sum of kg of products
+        """
+        self.tonnage = 0.0
+        for q in self.report_entries:
+            self.tonnage += q.tonnage
+
+    @api.one
+    @api.depends('parent_project')
+    def compute_user_can_moderate(self):
+        """
+        Is current user authorized to moderate (accept/reject) the detailed_report?
+        """
+        self.user_can_moderate = (
+            self.parent_project.manager.id == self.env.user.id or
+            self.parent_project.organization.coordinator.id == self.env.user.id
+        )
+
+    @api.one
+    @api.constrains('report_entries')
+    def check_report_entries(self):
+        listlen = len(self.report_entries)
+        newlist = set((i.commodity_code, i.detailed_report) for i in self.report_entries)
+        if (listlen > len(newlist)):
+            raise exceptions.ValidationError("Produkt o danym kodzie może mieć co najwyżej jeden wpis!")
+
+    @api.one
+    @api.depends('project')
+    def compute_project_dates(self):
+        """
+        For nice format in the list of all projects.
+        """
+        if self.project:
+            tmp_date = parser.parse(self.project.date_start).strftime("%d-%m-%Y")
```

Comment:
Odoo has a built-in method for converting their datetime fields values to python datetime objects. Don't parse them by hand - dealing with time is scary enough ;) Odoo's built-in method is well tuned for dealing with the format they use.

``` 
fields.Datetime.fromstring(self.project.datetime).strftime("%d-%m-%Y")
```